### PR TITLE
MB-10110 display NTSR specific information shipment display cards

### PR DESCRIPTION
**Pull request by mlm55**
## Jira tickets for this change
- [MB-10110](https://dp3.atlassian.net/browse/MB-10110) 
- [MB-10114](https://dp3.atlassian.net/browse/MB-10114) 
- [MB-10104](https://dp3.atlassian.net/browse/MB-10104) 

## Summary
This PR adds an nts release specific shipment list. It also adds error/warning flags for missing information, puts the correct label on shipment cards, and disables the submit when specific fields are missing.

Is there anything you would like reviewers to give additional scrutiny?
I created a new shipment list to handle ntsr shipments rather than adding a bunch of conditionals into the already existing shipment display list. Was that the right way to go?

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. On storybook look at http://localhost:6006/?path=/story/office-components-shipment-display--nts-release-shipment to see a shipment that has all required fields 
2.  On storybook look at http://localhost:6006/?path=/story/office-components-shipment-display--nts-release-shipment-missing-info to see a shipment that has missing fields
2. I don't think there's a storybook for the move details page, so check in the office app
2. Login as a service counselor
3. Find a move that already exists in your DB
4. Use postman/or manually edit the database to create an NTS release shipment that's missing the storage facility information and shipment weight
```
{
    "moveTaskOrderID": "use-a-move-id-in-your-db",
    "requestedPickupDate": "2022-12-12",
    "destinationAddress": {
    "streetAddress1": "123 Main Ave",
    "city": "Anytown",
    "state": "DE",
    "postalCode": "90210",
    "country": "USA"
  },
    "shipmentType": "HHG_OUTOF_NTS_DOMESTIC",
    "tacType": "HHG",
    "sacType": "NTS",
    "usesExternalVendor": true,
    "diversion": true,
}
```
5. Go to the move details page and check that the storage facility has an error flag and that the submit button is disabled

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

### CODE REVIEW

File: src/components/Office/DefinitionLists/NTSRShipmentInfoList.test.jsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    expect(customerRemarks).toHaveTextContent(info.customerRemarks);
+
+    const tacType = screen.getByTestId('tacType');
+    expect(tacType).toHaveTextContent('1234 (HHG)');
+
+    const sacType = screen.getByTestId('sacType');
+    expect(sacType).toHaveTextContent('1234123412 (NTS)');
+  });
+
+  it('renders a dash and adds a warning class for non-required missing items', () => {
```
Comment (YanZ777) [inline src/components/Office/DefinitionLists/NTSRShipmentInfoList.test.jsx:95]: Can we convert this to be table driven as well?

File: src/components/Office/DefinitionLists/ShipmentDefinitionLists.module.scss [hunk 2]
```diff
@@ -0,0 +1,65 @@
+@import 'shared/styles/_basics';
+@import 'shared/styles/colors';
+@import 'shared/styles/_mixins';
+@import 'shared/styles/_variables';
+
+.ShipmentDefinitionLists {
```
Comment (YanZ777) [inline src/components/Office/DefinitionLists/ShipmentDefinitionLists.module.scss:6]: Are there styles that can be reused from `/src/styles/descriptionList.module.scss` instead? Or maybe combining the two?

**Review by YanZ777 — CHANGES_REQUESTED** [review]
My comments are mainly for polishing up the tests, but everything else looks fine to me.

**Review by amanda5iu — COMMENTED** [review]
Facility info is required in addition to facility address - specifically the facility name. So facility info should be displayed in [this view](http://localhost:6006/?path=/story/office-components-shipment-display--nts-release-shipment).

There should also be 2 units (16px) of space to the left and above the `Edit` button.

image

**Edit:** my first comment is incorrect! Do not show facility info in the collapsed view of the card. Also, the padding should be 24px between the list and `Edit` button, not 16px.

File: src/components/Office/DefinitionLists/NTSRShipmentInfoList.jsx [hunk 3]
```diff
@@ -0,0 +1,206 @@
+import React from 'react';
+import * as PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import styles from './ShipmentDefinitionLists.module.scss';
+
+// import styles from 'styles/descriptionList.module.scss';
```
Comment (YanZ777) [inline src/components/Office/DefinitionLists/NTSRShipmentInfoList.jsx:7]: You might have forgotten to take this out?

**Review by YanZ777 — APPROVED** [review]
Changes look good to me!

File: src/components/Office/DefinitionLists/NTSRShipmentInfoList.test.jsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    ['agent', shipment.agents[0].email, { exact: false }],
+    ['counselorRemarks', shipment.counselorRemarks],
+    ['customerRemarks', shipment.customerRemarks],
+    ['tacType', '1234 (HHG)'],
+    ['sacType', '1234123412 (NTS)'],
+  ])('Verify Shipment field %s with value %s is present', async (shipmentField, shipmentFieldValue) => {
+    render(<NTSRShipmentInfoList isExpanded shipment={shipment} />);
+    const shipmentFieldElement = screen.getByTestId(shipmentField);
+    expect(shipmentFieldElement).toHaveTextContent(shipmentFieldValue);
+  });
```
Comment (donaldthai) [inline src/components/Office/DefinitionLists/NTSRShipmentInfoList.test.jsx:75]: I see we're not using any `await` call for some tests. Can we just remove the `async` keyword?

File: src/components/Office/ShipmentDisplay/ShipmentDisplay.jsx [hunk 5]
```diff
@@ -47,9 +127,9 @@ const ShipmentDisplay = ({ shipmentType, displayInfo, onChange, shipmentId, isSu
             )}
           </div>
 
-          <FontAwesomeIcon icon="chevron-down" />
+          <FontAwesomeIcon className={styles.icon} icon={expandableIconClasses} onClick={handleExpandClick} />
```
Comment (donaldthai) [inline src/components/Office/ShipmentDisplay/ShipmentDisplay.jsx:130]: I feel like clicking on the icon to expand the fields is a bit small. What if we set the onClick to the entire heading? Maybe @amanda5iu has more of an opinion on this?

**Review by donaldthai — APPROVED** [review]
Code looks good 😃 . I left some non-blocking comments.

File: src/components/Office/ShipmentDisplay/ShipmentDisplay.jsx (hunk 5 above)
Comment (amanda5iu) [reply src/components/Office/ShipmentDisplay/ShipmentDisplay.jsx:130]: @donaldthai agreed! There were specs for this somewhere when we first developed this pattern for HHG, but I can't find them at the moment. The whole header section should be clickable. I know this isn't in scope for this PR though!

**Review by amanda5iu — APPROVED** [review]

**Comment by donaldthai** [comment]
I approved the Happo changes since it seem like a timing issue.

**Pull request merged**